### PR TITLE
Generation of libkiwix.pc via meson's pkgconfig module

### LINKED ISSUE #1117: Improvement pkgconfig libkiwix
Why the naming is different? We have `kiwix` and `libzim`:
```
/usr/local/lib/x86_64-linux-gnu/pkgconfig/kiwix.pc
/usr/local/lib/x86_64-linux-gnu/pkgconfig/libzim.pc
```

It should not be `libkiwix.pc`?

Why we do have `kiwix.pc.in` in our code repo, but nothing similar at libzim level?

### PR DESCRIPTION
**Pull request by veloman-yunkan**
Fixes #1117

The name of the `libkiwix` pkgconfig file has changed from `kiwix.pc` to `libkiwix.pc`.

Below are the contents of libkiwix.pc and kiwix.pc files generated using the new and old approaches, respectively, for `native_dyn` and `native_static` configurations:

```
$ cat BUILD_native_dyn/INSTALL/lib/x86_64-linux-gnu/pkgconfig/libkiwix.pc
prefix=<REDACTED>
includedir=${prefix}/include
libdir=${prefix}/lib/x86_64-linux-gnu

Name: libkiwix
Description: A library that contains useful primitives that Kiwix readers have in common
Version: 14.0.0
Requires.private: icu-i18n, libzim < 10.0.0, libzim >= 9.0.0, pugixml, libcurl, libmicrohttpd, zlib, xapian-core
Libs: -L${prefix}/lib/x86_64-linux-gnu -lkiwix
Libs.private: -pthread
Cflags: -I${includedir}

$ cat BUILD_native_dyn/INSTALL/lib/x86_64-linux-gnu/pkgconfig/kiwix.pc
prefix=<REDACTED>
libdir=${prefix}/lib64
includedir=${prefix}/include

Name: libkiwix
Description: A library that contains a lot of things used by used by other kiwix programs
Version: 14.0.0
Requires: libzim icu-i18n pugixml libcurl libmicrohttpd xapian-core
Libs: -L${libdir} -lkiwix
Cflags: -I${includedir}/

$ cat BUILD_native_static/INSTALL/lib/x86_64-linux-gnu/pkgconfig/libkiwix.pc
prefix=<REDACTED>
includedir=${prefix}/include
libdir=${prefix}/lib/x86_64-linux-gnu

Name: libkiwix
Description: A library that contains useful primitives that Kiwix readers have in common
Version: 14.0.0
Requires: icu-i18n, libzim < 10.0.0, libzim >= 9.0.0, pugixml, libcurl, libmicrohttpd, zlib, xapian-core
Libs: -L${prefix}/lib/x86_64-linux-gnu -lkiwix -pthread
Cflags: -I${includedir} -pthread

$ cat BUILD_native_static/INSTALL/lib/x86_64-linux-gnu/pkgconfig/kiwix.pc
prefix=<REDACTED>
libdir=${prefix}/lib64
includedir=${prefix}/include

Name: libkiwix
Description: A library that contains a lot of things used by used by other kiwix programs
Version: 14.0.0
Requires: libzim icu-i18n pugixml libcurl libmicrohttpd xapian-core
Libs: -L${libdir} -lkiwix
Cflags: -I${includedir}/
```

The notable differences are:

- `libdir` changed from `${prefix}/lib64` to `${prefix}/lib/x86_64-linux-gnu`

- for `native_dyn` configuration `Requires.private` is used

- `pthread` has appeared in `Libs`/`Libs.private` and/or `Cflags` (likely inherited from `libzim`)

- version information was added to the `libzim` requirement.

### CODE REVIEW

**Comment by kelson42** [comment]
@veloman-yunkan Great, will do my best to check this PR. I believe we probably need to have PR at kiwix-tools and kiwix-desktop tomsupport this new naming?

File: meson.build [hunk 1]
```diff
@@ -35,8 +35,7 @@ else
   error('Cannot found header mustache.hpp')
 endif
 
-libzim_dep = dependency('libzim', version:'>=9.0.0', static:static_deps)
-libzim_dep = dependency('libzim', version:'<10.0.0', static:static_deps)
+libzim_dep = dependency('libzim', version:['>=9.0.0', '<10.0.0'], static:static_deps)
 
 if not compiler.has_header_symbol('zim/zim.h', 'LIBZIM_WITH_XAPIAN', dependencies: libzim_dep)
   error('Libzim seems to be compiled without xapian. Xapian support is mandatory.')
```
Comment (kelson42) [inline meson.build:41]: We should write "Xapian" not "xapian".

File: meson.build [hunk 2]
```diff
@@ -75,17 +74,10 @@ if get_option('doc')
...
-               input : 'kiwix.pc.in',
-               install_dir: get_option('libdir')+'/pkgconfig'
-              )
-
+pkg_mod = import('pkgconfig')
+pkg_mod.generate(libraries : [libkiwix] + extra_libs,
+                 version : meson.project_version(),
+                 name : 'libkiwix',
+                 filebase : 'libkiwix',
+                 description : 'A library that contains a lot of things used by other kiwix programs',
```
Comment (kelson42) [inline meson.build:82]: I propose a small rephrasing to "A library that contains useful primitives that Kiwix readers have in common".

**Review by kelson42 — APPROVED** [review]
LGTM modulo two small cosmetic changes

**Comment by veloman-yunkan** [comment]
> I believe we probably need to have PR at kiwix-tools and kiwix-desktop tomsupport this new naming?

@kelson42 Here they are - kiwix/kiwix-tools#707, kiwix/kiwix-desktop#1213

File: meson.build (hunk 1 above)
Comment (veloman-yunkan) [reply meson.build]: Done

File: meson.build (hunk 2 above)
Comment (veloman-yunkan) [reply meson.build]: It was a risky change but I managed to do it :)

**Comment by veloman-yunkan** [comment]
>   * version information was added to the `libzim` requirement (however,  for some reason it states `>= 8.1.0`, which is a deviation from the  specified version restriction of `['>=9.0.0', '<10.0.0']`; I was  unable to find out where it comes from).

The old version information appears to be a remnant of the initial setup of the build tree long ago with further utilization of a quick build procedure without using `kiwix-build`. After I ran a `kiwix-build` initiated build of `libkiwix` the version information of `libzim` was updated. So no concern here.

**Comment by kelson42** [comment]
@veloman-yunkan Should we not remove `kiwix.pc.in`?

**Comment by veloman-yunkan** [comment]
> @veloman-yunkan Should we not remove `kiwix.pc.in`?

@kelson42 Thanks for catching this. `kiwix.pc.in` is now history.

**Pull request merged**